contracts/Vault.sol

```diff
@@ -354,6 +356,9 @@ contract Vault is
         // disinvest
         if (alreadyInvested > maxInvestableAmount) {
             uint256 disinvestAmount = alreadyInvested - maxInvestableAmount;
+
+            require(disinvestAmount >= 10 * 10 ** underlying.decimals(), "Vault: Not enough to disinvest");
```

Comment:
we should put this value into an `immutable` variable in order to avoid the extra cross-contract call every time

Reply:
because underlying token is determined in constructor during deployment can we compromise with a private variable so that we can set it once in the constructor?

like this a vault deployed for USDC does not need another commit to support the 6 decimals instead of 18

Reply:
That's exactly what I was suggesting, just with the addition of the immutable keyword 😛
Being private or public doesn't matter much here

Reply:
gotcha, implemented :+1: